projects/RabbitMQ.Client/Impl/Channel.PublisherConfirms.cs

```diff
@@ -295,7 +295,10 @@ await _confirmSemaphore.WaitAsync(cancellationToken)
                 if (_publisherConfirmationTrackingEnabled)
                 {
                     publisherConfirmationTcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
-                    _confirmsTaskCompletionSources[publishSequenceNumber] = publisherConfirmationTcs;
+                    if (!_confirmsTaskCompletionSources.TryAdd(publishSequenceNumber, publisherConfirmationTcs))
+                    {
+                        throw new InvalidOperationException($"Failed to track the publisher confirmation for sequence number '{publishSequenceNumber}' because it already exists.");
```

Comment:
Please don't hesitate to suggest a different message and/or exception type

Reply:
Fine with me!